dask/distributed/scheduler.py

```diff
@@ -539,9 +543,9 @@ def _setitem_ack(self, header, payload):
 
     def close_workers(self):
         header = {'function': 'close'}
-        for w in self.workers:
+        while self.workers != {}:
```

Comment:
I need to iterate over the `self.workers` dict and destroy it as I go. I do this instead of a for-lop since you can't change a dict in a for-loop if you are iterating over it.